recipes/sfml/all/conanfile.py

```diff
@@ -0,0 +1,200 @@
+            self.requires('openal/1.19.1')
+            self.requires('flac/1.3.3')
+            self.requires('ogg/1.3.4')
+            self.requires('vorbis/1.3.6')
+        if self.options.window:
+            if self.settings.os in ['Linux', 'FreeBSD']:
+                self.requires('xorg/system')
+            self.requires('opengl/system')
+
+    def system_requirements(self):
+        if self.settings.os == 'Linux' and tools.os_info.is_linux:
+            if tools.os_info.with_apt:
+                installer = tools.SystemPackageTool()
+                packages = []
+                if self.options.window:
+                    packages.extend(['libudev-dev'])
+                for package in packages:
+                    installer.install(package)
+
+    def configure(self):
+        if self.options.shared:
+            del self.options.fPIC
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version])
+        extracted_dir = 'SFML-' + self.version
+        os.rename(extracted_dir, self._source_subfolder)
+        tools.rmdir(os.path.join(self._source_subfolder, "extlibs"))
+
+    def _configure_cmake(self):
+        if self._cmake:
+            return self.cmake
+        self.cmake = CMake(self)
+
+        self.cmake.definitions['SFML_BUILD_WINDOW'] = self.options.window
+        self.cmake.definitions['SFML_BUILD_GRAPHICS'] = self.options.graphics
+        self.cmake.definitions['SFML_BUILD_NETWORK'] = self.options.network
+        self.cmake.definitions['SFML_BUILD_AUDIO'] = self.options.audio
+
+        self.cmake.definitions['SFML_INSTALL_PKGCONFIG_FILES'] = False
+        self.cmake.definitions['SFML_GENERATE_PDB'] = False
+
+        self.cmake.configure(build_folder=self._build_subfolder)
+        return self.cmake
+
+    def build(self):
+        for patch in self.conan_data.get("patches",{}).get(self.version, []):
+            tools.patch(**patch)
+        cmake = self._configure_cmake()
+        cmake.build()
+
+    def package(self):
+        self.copy("license.md", dst="licenses", src=self._source_subfolder)
+        cmake = self._configure_cmake()
+        cmake.install()
+
+        tools.rmdir(os.path.join(self.package_folder, "lib", "cmake"))
+        os.remove(os.path.join(self.package_folder, "license.md"))
+        os.remove(os.path.join(self.package_folder, "readme.md"))
+
+    def _get_decorated_lib(self, name):
+        suffix = '-s' if not self.options.shared else ''
+        suffix += '-d' if self.settings.build_type == 'Debug' else ''
+        return name + suffix
+
+    def package_info(self):
+
+        self.cpp_info.names["cmake_find_package"] = "SFML"
+        self.cpp_info.names["cmake_find_package_multi"] = "SFML"
+        self.cpp_info.names["pkg_config"] = "SFML"
+
+        self.cpp_info.components["System"].names["cmake_find_package"] = "system"
```

Comment:
Components name could be lower case, it would avoid to override `cmake_find_package` and `cmake_find_package_multi`.
Personal opinion: since recipes names are always lower case in CCI, I think that always use lowercase for components is more consistent.

Reply:
By the way, SFML module and config files cannot be properly modeled by conan generators.
Usage is:

```cmake
find_package(SFML REQUIRED main window audio CONFIG)
target_link_libraries(myapp sfml-main sfml-window sfml-audio)
```